src/EditorFeatures/Core/Structure/StructureTag.cs

```diff
@@ -59,6 +61,31 @@ public StructureTag(AbstractStructureTaggerProvider tagProvider, BlockSpan block
         public bool IsDefaultCollapsed { get; }
         public bool IsImplementation { get; }
 
+        // Intentionally returning the base impl, we have never supported this facility, and there is no contract around
+        // placing these tags in sets or maps.
+        //
+        // Note: editor does use this.  But we've never supported this, so keeping default behavior unless we are told
+        // it is necessary to provide this:
+        // https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_git/VS-Platform?path=/src/Editor/Text/Impl/Structure/StructureSpanningTree/StructureSpanningTree.cs&version=GBmain&line=308&lineEnd=309&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents
+        public override int GetHashCode()
+            => base.GetHashCode();
```

Comment:
This is really terrifying, since this could easily break whatever the Editor is doing here; I'd expect that we provide a correct implementation (even if it's Jared's "famous" implementation of `return 42;`) until that editor code doesn't exist.

Reply:
just to be clear.  we were "broken" before in that we just hash randomly.